framework/src/partitioner/LibmeshPartitioner.C

```diff
@@ -123,14 +132,42 @@ return libmesh_make_unique<MetisPartitioner>();
 }
 
 void
+LibmeshPartitioner::prepare_blocks_for_subdomain_partitioner(MeshBase & /*mesh*/, SubdomainPartitioner & subdomain_partitioner)
+{
+  auto group_begin = _subdomain_blocks.begin();
+  auto group_end = _subdomain_blocks.end();
+
+  subdomain_partitioner.chunks.clear();
+  for (auto group = group_begin; group != group_end; ++group)
+  {
+     auto subdomain_begin = group->begin();
+     auto subdomain_end = group->end();
+     std::set<subdomain_id_type> subdomain_ids;
+     for (auto subdomain_name = subdomain_begin; subdomain_name != subdomain_end; ++subdomain_name)
+     {
+       // How to get a subdomain id by its name?
+       // mesh.get_id_by_name(*subdomain_name) does not work
+       subdomain_ids.insert(std::stoi(*subdomain_name));
```

Comment:
You will need to add it into `MoosePartitioner` input parameters in the `PartitionerAction`. Go look at `CreateProblemAction` to see how to do it.